app/code/Magebit/Faq/Block/Adminhtml/Faq/Edit/GenericButton.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * This file is part of the Magebit package.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magebit Faq
+ * to newer versions in the future.
+ *
+ * @copyright Copyright (c) 2024 Magebit, Ltd. (https://magebit.com/)
+ * @license   GNU General Public License ("GPL") v3.0
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Magebit\Faq\Block\Adminhtml\Faq\Edit;
+
+use Magento\Backend\Block\Widget\Context;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\UrlInterface;
+use Magento\Cms\Api\BlockRepositoryInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class GenericButton
+ *
+ * Provides generic button functionalities for the FAQ edit form, including generating URLs and retrieving block IDs.
+ *
+ * @package Magebit\Faq\Block\Adminhtml\Faq\Edit
+ */
+class GenericButton
+{
+    /**
+     * @param Context $context
+     * @param UrlInterface $url
+     * @param BlockRepositoryInterface $blockRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        private readonly Context $context,
+        private readonly UrlInterface $url,
+        private readonly BlockRepositoryInterface $blockRepository,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * Return the CMS block ID
+     *
+     * @return int|null
+     * @throws LocalizedException
```

Comment:
If you use try and catch, you can remove the throws. but here it looks like you are handling nosuchentityexception and not localizedexception. Perhaps adding it to catch would fix the issue. like catch (NoSuchEntityException|LocalizedException $e) ...

Do this in all places. 
I would generally recommend not using throws as it breaks the code when something goes wrong. A good practice is to handle the exception correctly by logging the error in the log files, if needed add a message for the user and then returning/ redirecting or whatever needs to be done to not break the code